### PR TITLE
[v7r3] Python 3 hackathon fixes

### PR DESCRIPTION
**Pull request by TaykYoku**
BEGINRELEASENOTES
Python 3 hackathon fixes.

*Core
CHANGE: modify MySQL._to_string

*WorkloadManager
FIX: decode values using _to_string method in JobDB
FIX: mock _escapeString in Test_Agent_PilotStatusAgent to pass test
FIX: escape values in PilotAgentsDB
CHANGE: remove pilots before start Test_PilotsClient test

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 1]
```diff
@@ -217,7 +217,7 @@ def export_getJobPilotOutput(self, jobID):
             c = cycle
 
     if pilotReference:
-      return getGridJobOutput(pilotReference)
+      return getGridJobOutput(pilotReference.decode())
```
Comment (chrisburr) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:220]: This should be fixed a level deeper in one or more of: `getAtticJobParameters`/`getJobParameter`/`getJobParameters`

Comment (chrisburr) [reply src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:220]: @TaykYoku has debugged this further and got to the bottom of it on slack: https://lhcbdirac.slack.com/archives/C3AGWCA8J/p1625147097062700

In summary, the JobParameters table is using `BLOB` for the value when it should be using a text type.

**Comment by TaykYoku** [comment]
restart tests

**Comment by TaykYoku** [comment]
re-test

**Comment by TaykYoku** [comment]
re-test

**Comment by TaykYoku** [comment]
rebased and ready for review.

NOTE: PR fixes do not affect the database structure, IMHO it can be considered separately from issue.

**Review by fstagni — DISMISSED** [review]

File: src/DIRAC/Core/Utilities/MySQL.py [hunk 2]
```diff
@@ -942,10 +942,14 @@ def _to_value(self, param):
...
-  def _to_string(self, param):
+  def _to_string(self, value):
     """
+      Convert value to string
     """
-    return param[0].tostring()
+    try:
+      return value.decode()
+    except (UnicodeDecodeError, AttributeError):
+      return str(value)
```
Comment (chrisburr) [inline src/DIRAC/Core/Utilities/MySQL.py:952]: If there is a `UnicodeDecodeError` then `return str(value)` is definitely the wrong thing to do as we'll get junk that starts to spread around. I'm also no sure when `AttributeError` would trigger (`value` is already a string maybe)?
```suggestion
    return value.decode()
```

Comment (chrisburr) [reply src/DIRAC/Core/Utilities/MySQL.py:952]: Looking further down this PR: I think `_to_string(xxx)` can be deleted in favour of using `xxx.decode()`

Comment (TaykYoku) [reply src/DIRAC/Core/Utilities/MySQL.py]: looking at the scope of this method, if only to convert a variable that is a string, then agree

**Review by atsareg — APPROVED** [review]

**Pull request merged**